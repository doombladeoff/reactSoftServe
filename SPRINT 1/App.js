import {
  StyleSheet,
  Image,
  View,
  SafeAreaView,
  Platform,
  StatusBar,
} from "react-native";

import ImagePile from "./components/ImagePile";
import StatisticsInfo from "./components/StatisticsInfo";

const backgroundImg = require("./assets/ocean.jpg");

export default function App() {
  return (
    <SafeAreaView style={styles.safeArea}>
      <Image
        source={backgroundImg}
        style={{ height: "100%", width: "100%", position: "absolute" }}
      />
      <View style={styles.container}>
        <StatisticsInfo />
        <ImagePile />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    marginTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
  },
  container: {
    flex: 1,
  },
});
