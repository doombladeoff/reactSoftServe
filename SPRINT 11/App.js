import { useEffect, useRef, useState } from "react";
import { View, StyleSheet, TextInput, SafeAreaView, Text } from "react-native";
import * as Location from "expo-location";
import { ModalAddress } from "./components/Modal";
import { MapViewer } from "./components/MapViewer";
import { Buttons } from "./components/Buttons";

const ZOOM_LEVEL = 15;
const ANIMATION_DURATION = 2000;

const ERROR_MESSAGE = {
    NOT_FOUND_ADDRESS: 'Address not found',
    NO_PERMISSION: 'Permission to access location was denied',
}

export default function App() {
    const [location, setLocation] = useState(null);
    const [address, setAddress] = useState('');
    const [marker, setMarker] = useState(null);
    const [coordinates, setCoordinates] = useState([]);
    const [isModalVisible, setIsModalVisible] = useState({ visible: false, errorText: '' });
    const mapRef = useRef(null);

    useEffect(() => {
        (async () => {
            let { status } = await Location.requestForegroundPermissionsAsync();
            if (status !== 'granted') {
                return setIsModalVisible({ visible: true, errorText: ERROR_MESSAGE.NO_PERMISSION });
            }

            let location = await Location.getCurrentPositionAsync({});
            setLocation(location);

            await centerMap(location.coords.latitude, location.coords.longitude);
        })();
    }, []);

    const addMarker = async () => {
        if (address.trim() === '') return;
        try {
            let findAddress = await Location.geocodeAsync(address);
            if (findAddress.length === 0) {
                return setIsModalVisible({ visible: true, errorText: ERROR_MESSAGE.NOT_FOUND_ADDRESS });
            }
            let { latitude, longitude } = findAddress[0];
            setMarker({ latitude, longitude });
            setCoordinates([
                {
                    latitude: location.coords.latitude,
                    longitude: location.coords.longitude
                },
                {
                    latitude,
                    longitude
                }
            ]);

            await centerMap(latitude, longitude);
        } catch (error) {
            setIsModalVisible({ visible: true, errorText: ERROR_MESSAGE.NOT_FOUND_ADDRESS });
        }
    };

    const centerMap = (latitude, longitude) => {
        mapRef.current.animateCamera({
            center: {
                latitude: latitude,
                longitude: longitude
            },
            zoom: ZOOM_LEVEL,
        }, { duration: ANIMATION_DURATION });
    };

    return (
        <SafeAreaView style={{ flex: 1, backgroundColor: 'transparent' }}>
            <View style={styles.container}>
                <View style={{ alignItems: 'center' }}>
                    <Text>MAP</Text>
                </View>
                <MapViewer
                    mapRef={mapRef}
                    marker={marker}
                    coordinates={coordinates}
                    location={location}
                />
                <TextInput
                    style={styles.input}
                    placeholder="Enter the address"
                    value={address}
                    onChangeText={setAddress}
                />
                <View style={styles.buttonContainer}>
                    <Buttons
                        onPress={addMarker}
                        textButton={'Add marker'}
                    />
                    {marker ? <Buttons onPress={() => centerMap(marker.latitude, marker.longitude)}
                                       textButton={'Center the marker'}
                    /> : null}
                </View>
                {isModalVisible &&
                    <ModalAddress
                        isShow={isModalVisible}
                        onClose={() => {
                            setIsModalVisible({ visible: false, errorText: '' });
                        }}
                    />
                }
            </View>
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "flex-end",
    },
    input: {
        width: "100%",
        height: 40,
        backgroundColor: "white",
        borderWidth: 1,
        borderColor: "#ccc",
        paddingHorizontal: 10,
        marginBottom: 10,
    },
    buttonContainer: {
        flexDirection: "row",
        justifyContent: "space-evenly",
        width: "100%",
        alignItems: "center",
        paddingHorizontal: 10,
        marginBottom: 20,
    },
})

