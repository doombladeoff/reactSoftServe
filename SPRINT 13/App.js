import React, { useCallback } from "react";
import { StyleSheet, Text, View, Dimensions, Pressable } from "react-native";
import Animated, {
    useSharedValue,
    withTiming,
    useAnimatedProps,
    useDerivedValue,
} from "react-native-reanimated";
import { ReText } from "react-native-redash";

import Svg, { Circle } from "react-native-svg";

const BACKGROUND_COLOR = "#F0EBED";
const BACKGROUND_STROKE_COLOR = "#f4b7d7";
const BACKGROUND_BUTTON_COLOR = "#c00069";
const BACKGROUND_BUTTON_PRESSED_COLOR = "#A3025A";
const STROKE_COLOR = "#e91f8b";

const { width, height } = Dimensions.get("window");

const CIRCLE_LENGTH = 1000;
const R = CIRCLE_LENGTH / (2 * Math.PI);

export default function App() {
    const AnimatedCircle = Animated.createAnimatedComponent(Circle);
    const progress = useSharedValue(0);
    const animatedProps = useAnimatedProps(() => ({
        strokeDasharray: [progress.value * CIRCLE_LENGTH, (1 - progress.value) * CIRCLE_LENGTH],
    }));

    const progressText = useDerivedValue(() => {
        return `${Math.floor(progress.value * 100)}`;
    });

    const startAnimation = useCallback(() => {
        progress.value = withTiming(progress.value === 0 ? 1 : 0, { duration: 2000 });
    }, [progress.value]);


    return (
        <View style={styles.container}>
            <Svg style={{ position: "absolute" }}>
                <Circle
                    cx={width / 2}
                    cy={height / 2}
                    r={R}
                    stroke={BACKGROUND_STROKE_COLOR}
                    strokeWidth={30}
                />
                <AnimatedCircle
                    cx={width / 2}
                    cy={height / 2}
                    r={R}
                    stroke={STROKE_COLOR}
                    strokeWidth={25}
                    strokeDashoffset={250}
                    animatedProps={animatedProps}
                    strokeLinecap={"butt"}
                    fill={BACKGROUND_COLOR}
                />
            </Svg>
            <ReText
                style={styles.progressText}
                text={progressText}
            />
            <Pressable
                onPress={startAnimation}
                style={({ pressed }) => [
                    styles.button,
                    {
                        backgroundColor: pressed
                            ? BACKGROUND_BUTTON_PRESSED_COLOR
                            : BACKGROUND_BUTTON_COLOR,
                    },
                ]}
            >
                <Text style={styles.buttonText}>Run</Text>
            </Pressable>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: BACKGROUND_COLOR,
        alignItems: "center",
        justifyContent: "center",
    },
    progressText: {
        fontSize: 80,
        color: "#3C3B3B",
        width: 200,
        textAlign: "center",
    },
    button: {
        position: "absolute",
        bottom: 80,
        width: width * 0.7,
        height: 60,
        borderRadius: 15,
        elevation: 5,
        alignItems: "center",
        justifyContent: "center",
    },
    buttonText: {
        fontSize: 25,
        color: "white",
        letterSpacing: 2.0,
    },
});
