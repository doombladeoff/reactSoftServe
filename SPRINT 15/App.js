import { StatusBar } from "expo-status-bar";
import { StyleSheet } from "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";

import { Colors } from "./utils/styles";
import LoginScreen from "./screens/LoginScreen";
import SignupScreen from "./screens/SignupScreen";
import WelcomeScreen from "./screens/WelcomeScreen";
import IconButton from "./components/ui/IconButton";
import { authActions } from "./store/authSlice";

import { Provider, useDispatch, useSelector } from "react-redux";
import store from "./store";
import { logout } from "./utils/authDataService";

const Stack = createNativeStackNavigator();

function AuthStack() {
    return (
        <Stack.Navigator
            screenOptions={{
                headerStyle: { backgroundColor: Colors.primary500 },
                headerTintColor: "white",
                contentStyle: { backgroundColor: Colors.primary100 },
            }}
        >
            <Stack.Screen name="Login" component={LoginScreen}/>
            <Stack.Screen name="Signup" component={SignupScreen}/>
        </Stack.Navigator>
    );
}

function EnteredStack() {
    const dispatch = useDispatch();
    const username = useSelector((state) => state.auth.login);
    const refreshToken = useSelector((state) => state.auth.refreshToken);
    return (
        <Stack.Navigator
            screenOptions={{
                headerStyle: { backgroundColor: Colors.primary500 },
                headerTintColor: "white",
                contentStyle: { backgroundColor: Colors.primary100 },
            }}
        >
            <Stack.Screen
                name={`Welcome to posts, ${username}!`}
                component={WelcomeScreen}
                options={{
                    headerRight: ({ tintColor }) => (
                        <IconButton
                            icon="exit"
                            color={tintColor}
                            size={24}
                            onPress={async () => {
                                try {
                                    await logout(refreshToken);
                                    dispatch(authActions.logout(refreshToken));
                                } catch (err) {
                                    console.log(err);
                                }
                            }
                            }

                        />
                    ),
                }}
            />
        </Stack.Navigator>
    );
}

function InnerNavigation() {
    const isAuthenticated = useSelector((state) => state.auth.isAuthenticated);
    return isAuthenticated ? <EnteredStack/> : <AuthStack/>;
}

export default function App() {
    return (
        <>
            <Provider store={store}>
                <StatusBar style="light"/>
                <NavigationContainer>
                    <InnerNavigation/>
                </NavigationContainer>
            </Provider>
        </>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#fff",
        alignItems: "center",
        justifyContent: "center",
    },
});
