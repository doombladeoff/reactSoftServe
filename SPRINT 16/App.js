import { StatusBar } from "expo-status-bar";
import React, { useState, useEffect } from "react";
import { Entypo } from '@expo/vector-icons';
import DateTimePicker from "@react-native-community/datetimepicker";
import {
    FlatList,
    Keyboard,
    StyleSheet,
    Text,
    TextInput,
    TouchableOpacity,
    View,
    Alert,
} from "react-native";
import uuid from "react-native-uuid";
import Countdown from "./components/Countdown";
import { services } from "./services/notifications";
import * as Notifications from "expo-notifications";

// ----- setup the notifications to works on the device here
async function setupNotifications() {
    try {
        await services.setup();
    } catch (error) {
        console.log('Error in setupNotifications:', error);
    }
};

export default function App() {
    const [tasks, setTasks] = useState([]);
    const [inputValue, setInputValue] = useState("");
    const [text, setText] = useState("");
    const [emoji, setEmoji] = useState("");
    const [inputDate, setDate] = useState(new Date());

    useEffect(() => {
        async function setup() {
            await setupNotifications();
            const subscription = await Notifications.addNotificationResponseReceivedListener(response => {
                const { actionIdentifier, notification } = response;
                if (actionIdentifier === 'delete') {
                    const { id } = notification.request.content.data;
                    deleteFromList(id).then(() => console.log(`delete item ${id}`));
                }
            });

            return () => subscription.remove();
        }

        setup();

    }, []);

    const handleInputChange = (value) => {
        setInputValue(value);
        separateTextAndEmoji(value);
    };

    const separateTextAndEmoji = (value) => {
        const emojiRegex = /(\p{Emoji_Presentation}|\p{Emoji}\uFE0F)/gu;
        const match = value.match(emojiRegex);
        const emojiPart = match ? match.join('') : '';
        const textPart = value.replace(emojiRegex, "").trim();
        setText(textPart);
        setEmoji(emojiPart);
    };

    const addToList = async () => {
        if (inputValue === "") {
            Alert.alert("Oops", "Input is empty", [{ text: "Ok" }]);
            return;
        }
        if (inputDate < new Date()) {
            Alert.alert("Oops", "Date must be future", [{ text: "Ok" }]);
            return;
        }

        const newItem = {
            id: uuid.v4(),
            title: text,
            emoji: emoji || "⏳", // ----- change the countdown icon here
            date: inputDate,
            notificationId: null
        };

        // ----- schedule the countdown notification here
        const getTriggerTime = ((inputDate.getTime() - new Date().getTime()) / 1000).toFixed(0);
        newItem.notificationId = await services.schedule(newItem.title, 'Countdown completed!', { id: newItem.id }, parseInt(getTriggerTime));

        setTasks([newItem, ...tasks]);
        setInputValue("");
        setDate(new Date());
        Keyboard.dismiss();

    };

    const deleteFromList = async (id) => {
        console.log('ID:', id)
        const item = tasks.find(i => i.id === id);

        console.log(item)
        console.log(tasks);
        if (item) {
            setTasks(tasks.filter((task) => task.id !== id));
            // ----- cancel the countdown notification here
            await services.cancel(item.notificationId);
        }
        console.log('NEW:', tasks)
    };

    return (
        <View style={styles.container}>
            <View style={styles.taskWrapper}>
                <View>
                    <Text style={styles.sectionTitle}>Countdowns</Text>
                </View>
                <TextInput
                    value={inputValue}
                    style={styles.input}
                    placeholder='Enter Countdown Title...'
                    testID="input"
                    onChangeText={(text) => handleInputChange(text)}
                />
                <View style={styles.datePicker}>
                    <DateTimePicker
                        value={inputDate}
                        mode="datetime"
                        testID="picker"
                        onChange={(e, date) => setDate(date)}
                    />
                </View>
                <TouchableOpacity
                    onPress={addToList}
                    testID="addBtn"
                >
                    <View style={styles.btn}>
                        <Text style={{ color: "white" }}>Start Countdown</Text>
                        <Entypo
                            style={{ marginLeft: 5, color: "white" }}
                            name='hour-glass'
                            size={20}
                            color='black'
                        />
                    </View>
                </TouchableOpacity>
                <View style={styles.items}>
                    <FlatList
                        keyExtractor={(item) => item.id}
                        data={tasks}
                        showsVerticalScrollIndicator={false}
                        renderItem={({ item }) => (
                            <Countdown countdown={item} deleteFromList={deleteFromList}/>
                        )}
                    />
                </View>
            </View>
            <StatusBar style='auto'/>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        height: "100%",
        backgroundColor: "#E8EAED",
    },
    btn: {
        flexDirection: "row",
        padding: 10,
        alignItems: "center",
        justifyContent: "center",
        borderRadius: 10,
        marginTop: 10,
        backgroundColor: "#008080",
        color: "#ffffff",
    },
    taskWrapper: {
        flex: 1,
        paddingTop: 80,
        paddingHorizontal: 20,
    },
    sectionTitle: {
        fontSize: 38,
        fontWeight: "bold",
        marginBottom: 10,
    },
    items: {
        flex: 1,
        marginTop: 30,
    },
    input: {
        backgroundColor: "#fff",
        padding: 15,
        borderRadius: 10,
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "space-between",
    },
    datePicker: {
        marginVertical: 10,
        flexDirection: "row",
        justifyContent: "center",
    },
});
