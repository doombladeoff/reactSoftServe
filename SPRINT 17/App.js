import React, { useState, useEffect } from "react";
import { View, TextInput, Button, StyleSheet, Text } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

import MealDisplay from "./components/MealDisplay";
import { fetchMeal } from "./services/mealService";

const App = () => {
  const [inputValue, setInputValue] = useState("");
  const [meal, setMeal] = useState(null);

  useEffect(() => {
    (async () => {
      setMeal(await fetchMeal(inputValue));
    })();
  }, []);

  const handleGetMeal = async () => {
    setMeal(await fetchMeal(inputValue));
    setInputValue("");
  };

  return (
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          value={inputValue}
          onChangeText={setInputValue}
          placeholder="Enter a meal name"
        />
        <Button title="Get Meal" onPress={handleGetMeal} />
        {meal ? (
          <MealDisplay meal={meal} />
        ) : (
          <Text style={styles.text}>No meal was found</Text>
        )}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: "white",
  },
  container: {
    flex: 1,
    padding: 20,
  },
  text: {
    paddingTop: 20,
    fontSize: 24,
    fontWeight: "bold",
    alignSelf: "center",
  },
  input: {
    height: 40,
    borderColor: "gray",
    borderWidth: 1,
    marginBottom: 20,
    paddingHorizontal: 10,
  },
});

export default App;
