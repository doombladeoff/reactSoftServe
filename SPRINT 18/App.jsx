import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import DateTimePicker from "react-native-date-picker";

const App = () => {
  const taskTitle = "Test task";
  const [taskDate, setTaskDate] = useState(null);
  const [selectedTask, setSelectedTask] = useState(false);
  const handleDateChange = (event, date) => {
    setTaskDate(date);
    setSelectedTask(false);
  };

  const handleTaskSelect = () => {
    if (selectedTask) return;
    setSelectedTask(true);
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.taskItem} onPress={handleTaskSelect}>
        <Text style={styles.taskTitle}>{taskTitle}</Text>
        <Text>{taskDate ? taskDate.toLocaleDateString() : "No Due Date"}</Text>
      </TouchableOpacity>
      {selectedTask && (
        <DateTimePicker
          testID="date-picker"
          value={taskDate || new Date()}
          mode="date"
          is24Hour={true}
          display="default"
          onChange={handleDateChange}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  taskItem: {
    minWidth: "80%",
    borderWidth: 1,
    borderColor: "lightgray",
    borderRadius: 5,
  },
  taskTitle: {
    fontSize: 16,
  },
});

export default App;
