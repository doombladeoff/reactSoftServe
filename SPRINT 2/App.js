import {
  StyleSheet,
  Text,
  View,
  Button,
  TextInput,
  Pressable,
} from "react-native";
import { useState } from "react";

const BACKGROUND_COLOR = "#ffffff";
const PRESSED_BACKGROUND_COLOR = "#ffcccc";
const NOTE_COLOR = "#ffffff";
const PRESSED_NOTE_COLOR = "#ffff00";

export default function App() {
  // your work with state
  const [inputText, setInputText] = useState("");
  const [notes, setNote] = useState([]);
  const [pressedNoteIndex, setPressedNoteIndex] = useState(null);

  function inputHandler(text) {
    return setInputText(text);
  }

  function add_note() {
    const isEmptyField = inputText.trim();
    if (!isEmptyField) {
      return;
    } else {
      setNote([...notes, inputText]), setInputText("");
    }
  }

  function longPress(index) {
    return (
      setPressedNoteIndex(index),
      alert("The note is pressed with a delay of 1 sec!")
    );
  }

  return (
    <View style={styles.appContainer}>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.textInput}
          placeholder="Enter your note"
          onChangeText={inputHandler}
          value={inputText}
        />
        <Button title="Add note" onPress={add_note} />
      </View>
      <View testID="notesContainer">
        {notes.map((note, index) => (
          <Pressable
            key={index}
            testID="pressableElem"
            onLongPress={() => longPress(index)}
            style={{
              backgroundColor:
                pressedNoteIndex === index
                  ? PRESSED_BACKGROUND_COLOR
                  : BACKGROUND_COLOR,
            }}
          >
            <Text
              testID="noteElem"
              style={[
                styles.noteElem,
                {
                  color:
                    pressedNoteIndex === index
                      ? PRESSED_NOTE_COLOR
                      : NOTE_COLOR,
                },
              ]}
            >
              {note}
            </Text>
          </Pressable>
        ))}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  appContainer: {
    paddingTop: 80,
    paddingHorizontal: 16,
  },
  inputContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingBottom: 28,
    marginBottom: 18,
    borderBottomWidth: 1,
    borderBottomColor: "#cccccc",
  },
  textInput: {
    borderWidth: 1,
    borderColor: "#cccccc",
    width: "70%",
    marginRight: 8,
    padding: 8,
  },
  noteElem: {
    margin: 8,
    padding: 8,
    borderRadius: 12,
    backgroundColor: "#008000",
    fontSize: 16,
    textAlign: "center",
  },
});
