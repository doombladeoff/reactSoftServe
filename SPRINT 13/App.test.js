import { Text } from "react-native";
import Animated, * as RestAnimated from "react-native-reanimated";
import * as ReDashModule from "react-native-redash";
jest.mock("react-native-reanimated");
jest.mock("react-native-redash");
import { render, screen, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import App from "./App";

function setupFunctions(progress) {
  Animated.createAnimatedComponent.mockImplementationOnce(
    () =>
      ({ animatedProps }) =>
        <Text>{JSON.stringify(animatedProps)}</Text>
  );
  RestAnimated.useAnimatedProps.mockImplementationOnce((argFunc) => argFunc());
  RestAnimated.useSharedValue.mockImplementationOnce(() => ({
    value: progress,
  }));
  RestAnimated.useDerivedValue.mockImplementationOnce(
    (argFunc) => argFunc()
  );
  RestAnimated.withTiming.mockImplementationOnce(() => 0.3);
  ReDashModule.ReText.mockImplementationOnce(({ text }) => <Text>{text}</Text>);
}
afterEach(() => {
  jest.resetAllMocks();
  jest.clearAllMocks();
});

test("when rpogress is 0 it is used correctly", async () => {
  const progress = 0;
  setupFunctions(progress);
  render(<App />);

  expect(RestAnimated.useSharedValue).toHaveBeenCalledWith(0);
  expect(RestAnimated.useAnimatedProps).toHaveBeenCalled();
  expect(RestAnimated.useDerivedValue).toHaveBeenCalled();
  expect(RestAnimated.withTiming).not.toHaveBeenCalled();
  const progressText = screen.getByText("0");
  const animatedPropsValue = screen.getByText(
    `{"strokeDasharray":[${0},${1000}]}`
  );
  expect(progressText).toBeOnTheScreen();
  expect(animatedPropsValue).toBeOnTheScreen();

  fireEvent.press(screen.getByText("Run"));
  expect(RestAnimated.withTiming).toHaveBeenCalledWith(1, { duration: 2000 });
});

test("when rpogress is 0.3 it is used correctly", async () => {
  const progress = 0.3;
  setupFunctions(progress);
  render(<App />);

  expect(RestAnimated.useSharedValue).toHaveBeenCalledWith(0);
  expect(RestAnimated.useAnimatedProps).toHaveBeenCalled();
  expect(RestAnimated.useDerivedValue).toHaveBeenCalled();
  expect(RestAnimated.withTiming).not.toHaveBeenCalled();
  const progressText = screen.getByText("30");
  const animatedPropsValue = screen.getByText(
    `{"strokeDasharray":[${progress * 1000},${(1 - progress) * 1000}]}`
  );
  expect(progressText).toBeOnTheScreen();
  expect(animatedPropsValue).toBeOnTheScreen();

  fireEvent.press(screen.getByText("Run"));
  expect(RestAnimated.withTiming).toHaveBeenCalledWith(0, { duration: 2000 });
});

test("when rpogress is 1 it is used correctly", async () => {
  const progress = 1;
  setupFunctions(progress);
  render(<App />);

  expect(RestAnimated.useSharedValue).toHaveBeenCalledWith(0);
  expect(RestAnimated.useAnimatedProps).toHaveBeenCalled();
  expect(RestAnimated.useDerivedValue).toHaveBeenCalled();
  expect(RestAnimated.withTiming).not.toHaveBeenCalled();
  const progressText = screen.getByText("100");
  const animatedPropsValue = screen.getByText(
    `{"strokeDasharray":[${1000},${0}]}`
  );
  expect(progressText).toBeOnTheScreen();
  expect(animatedPropsValue).toBeOnTheScreen();

  fireEvent.press(screen.getByText("Run"));
  expect(RestAnimated.withTiming).toHaveBeenCalledWith(0, { duration: 2000 });
});
test("when rpogress is 1 it is used correctly", async () => {
  setupFunctions(0);
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
