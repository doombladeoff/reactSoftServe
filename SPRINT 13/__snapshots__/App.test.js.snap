// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rpogress is 1 it is used correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F0EBED",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RNSVGSvgView
    bbHeight="100%"
    bbWidth="100%"
    focusable={false}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "position": "absolute",
        },
        {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGCircle
        cx="375"
        cy="667"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
          ]
        }
        r="159.15494309189535"
        stroke={
          {
            "payload": 4294227927,
            "type": 0,
          }
        }
        strokeWidth="30"
      />
      <Text>
        {"strokeDasharray":[0,1000]}
      </Text>
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text>
    0
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 15,
          "bottom": 80,
          "elevation": 5,
          "height": 60,
          "justifyContent": "center",
          "position": "absolute",
          "width": 525,
        },
        {
          "backgroundColor": "#c00069",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 25,
          "letterSpacing": 2,
        }
      }
    >
      Run
    </Text>
  </View>
</View>
`;
